src/control/lib/daos/api/handle.go

```diff
@@ -0,0 +1,92 @@
+//
+// (C) Copyright 2025 Google LLC
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package api
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+/*
+#include <daos.h>
+
+#cgo LDFLAGS: -ldaos_common
+*/
+import "C"
+
+const (
+	MissingPoolLabel      = "<no pool label supplied>"
+	MissingContainerLabel = "<no container label supplied>"
+)
+
+type (
+	// ctxHdlKey is a type used for storing handles as context values.
+	ctxHdlKey string
+
+	// connHandle is an opaque type used to represent a DAOS connection (pool or container).
+	connHandle struct {
+		UUID       uuid.UUID
+		Label      string
+		daosHandle C.daos_handle_t
+	}
+)
+
+// invalidate clears the handle so that it cannot be reused inadvertently.
+func (ch *connHandle) invalidate() {
+	if ch == nil {
+		return
+	}
+	ch.UUID = uuid.Nil
+	ch.Label = ""
+	ch.daosHandle.cookie = 0
+}
+
+// FillHandle copies the handle to the supplied pointer,
+// which must be a reference to a C.daos_handle_t.
+// NB: Caller is responsible for keeping the copy in sync with
+// this handle -- use of this method should be discouraged as
+// it is provided for compatibility with older code that calls
+// into libdaos directly.
+func (ch *connHandle) FillHandle(cHandle unsafe.Pointer) error {
```

Comment:
The caller is a CGo function, right? Any reason `cHandle` shouldn't explicitly be a `*C,daos_handle_t`?

Reply:
No, this function is for users outside of the API boundary (e.g. the daos tool). Can't use C types in public function signatures. Eventually we should be able to remove this entirely once the API is complete and all of the tool code is cleaned up.

Reply:
How does the caller end up with a C handle if it's not using C types? Just a piece that hasn't been translated to the API yet?

Reply:
> Just a piece that hasn't been translated to the API yet?

Yes, exactly. There's still a ton of code in the daos tool that does raw libdaos calls and those need pool/container handles. This helper lets us take advantage of the new pool and container connection management code while still working with the stuff that hasn't been migrated yet.

Reply:
Got it, thanks.